sdk_contrib/auto_config/src/main/java/io/opentelemetry/sdk/contrib/auto/config/SpanExporterFactory.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.contrib.auto.config;
+
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+
+/**
+ * A {@link SpanExporterFactory} acts as the bootstrap for a {@link SpanExporter} implementation. An
+ * exporter must register its implementation of a {@link SpanExporterFactory} through the Java SPI
```

Comment:
I am not sure about the SPI framework. In my understanding, the implementation should register the exporter via the API offered by the SDK.

Reply:
Well, this is for the auto instrumentation. The exporter and the auto instrumenter implementations don't know about each other, so we use SPI to load the exporter dynamically based on a JAR file name that's passed. 

Have a look here: https://github.com/open-telemetry/opentelemetry-auto-instr-java/blob/239eb53a44000f9f77e251a79e7189883451f5fc/agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java#L41

Reply:
Thanks for the clarification :+1: